### ISSUE TITLE
Support for resolution limits in config file to save bandwidth.

### ISSUE THREAD

**Issue by Fijxu** [iv-org/invidious-companion#26]
Should be easy. I have no problem in implementing it myself.

https://git.nadeko.net/Fijxu/invidious/commit/8f46bd57519c08f18f09166fe0c062921853099c

**Comment by unixfox** [comment]
In my opinion I would prefer to artifically limit the max resolution in video.js: https://github.com/videojs/http-streaming#vhsrepresentations

This way anyone can still select 1080p but by default for example only up to 720p will automatically be tested. This should greatly reduce the strain of the bandwidth used.

What do you think?

**Comment by Fijxu** [comment]
Good idea, but that would only limit the resolution on the Invidious front-end, not for every third party program that uses Invidious to fetch the videos (e.g: FreeTube, Clipious, yt-dlp for downloads). What I'm looking for with this pull request is a **hard** resolution limit, for everyone. IMO this is useful for public instance maintainers that have fixed bandwidth limits and low link speeds

**Comment by unixfox** [comment]
> Good idea, but that would only limit the resolution on the Invidious front-end, not for every third party program that uses Invidious to fetch the videos (e.g: FreeTube, Clipious, yt-dlp for downloads). What I'm looking for with this pull request is a **hard** resolution limit, for everyone. IMO this is useful for public instance maintainers that have fixed bandwidth limits and low link speeds

All the 3rd parties does not use the dash manifest API. They use the invidious API.

This means that we have to do some filtering on the video data given back to invidious. Once the API will be supported because right now it is not: https://github.com/iv-org/invidious-companion/issues/22

**Comment by Fijxu** [comment]
> This means that we have to do some filtering on the video data given back to invidious. Once the API will be supported because right now it is not: https://github.com/iv-org/invidious-companion/issues/22

I see

> PS: yt-dlp does not download from the invidious server, the tool just detects invidious and assume it's a youtube link then download from youtube servers directly.

I have seen people trying to use Invidious with yt-dlp to download videos since they can't download videos directly from youtube.
Reference: https://github.com/grqz/yt-dlp-invidious

**Comment by unixfox** [comment]
While looking at the video.js specification, I found something interesting to limit the bandwidth used: https://github.com/videojs/http-streaming/blob/main/docs/bitrate-switching.md

--------------

> > PS: yt-dlp does not download from the invidious server, the tool just detects invidious and assume it's a youtube link then download from youtube servers directly.
> 
> I have seen people trying to use Invidious with yt-dlp to download videos since they can't download videos directly from youtube. Reference: [grqz/yt-dlp-invidious](https://github.com/grqz/yt-dlp-invidious?rgh-link-date=2024-12-19T21%3A56%3A59Z)

IMO if it's an issue for you, you should:
- turn on "verify_requests": https://github.com/iv-org/invidious-companion/blob/master/config/default.toml. Any endpoints like latest_version and dash manifest won't work without a check generated through the frontend
- block the API requests
